src/platforms/common/swdptap.c

```diff
@@ -108,12 +117,11 @@ static uint32_t swdptap_seq_in_no_delay(const size_t clock_cycles)
 {
 	uint32_t value = 0;
 	for (size_t cycle = 0; cycle < clock_cycles; ++cycle) {
-		gpio_clear(SWCLK_PORT, SWCLK_PIN);
 		value |= gpio_get(SWDIO_IN_PORT, SWDIO_IN_PIN) ? 1U << cycle : 0U;
 		gpio_set(SWCLK_PORT, SWCLK_PIN);
 		__asm__("nop");
```

Comment:
This and the `nop` below could maybe do with being duplicated once to improve the resulting waveform, but this is only a suggestion as this does work and produce a valid minimum delay signal.

Reply:
I wound up making a different set of adjustments that similarly improve symmetry.

Reply:
That works! ^_^